pandas/core/groupby/generic.py

```diff
@@ -358,8 +357,10 @@ def _wrap_series_output(
         if len(output) > 1:
             result = DataFrame(indexed_output, index=index)
             result.columns = columns
-        else:
+        elif not columns.empty:
             result = Series(indexed_output[0], index=index, name=columns[0])
+        else:
```

Comment:
Do we need this branch? Is there a test that hits this?

Reply:
@WillAyd Here's a test that hits it:
```
pytest pandas/tests/groupby/aggregate/test_aggregate.py::TestNamedAggregationSeries::test_no_args_raises
```

Reply:
I was referring to the `result = DataFrame()` line - seems very strange to return this from this function

Reply:
In the test
```
pytest pandas/tests/groupby/aggregate/test_aggregate.py::TestNamedAggregationSeries::test_no_args_raises
```
we reach this branch. Before, it would hit the "else" branch, and would raise an error because `columns[0]` would be out of range - hence, I added this extra branch.

But we can avoid ever getting here by doing a check on `results` in `_aggregate_multiple_funcs`, so I'll do that instead

Reply:
@WillAyd how about returning `self.obj._constructor_expanddim()` - does that seem OK?